File: src/14-challenge/solution.js
/**
 * @param {number[]} reindeer
 * @param {number[]} stables
 * @returns {number}
 */
export default function minMovesToStables(reindeer, stables) {
	/** @type {number}>} */
	let minMoves = 0

	while (reindeer.length) {
		/** @type {number}>} */
		const minReinder = Math.min(...reindeer)

		/** @type {number}>} */
		const minStables = Math.min(...stables)

		minMoves += Math.abs(minReinder - minStables)

		reindeer.splice(reindeer.indexOf(minReinder), 1)
		stables.splice(stables.indexOf(minStables), 1)
	}

	return minMoves
}

File: src/14-challenge/solution.py
def min_moves_to_stables(reindeer: list[int], stables: list[int]) -> int:
    min_moves: int = 0

    while len(reindeer):
        min_reindeer: int = min(reindeer)
        min_stables: int = min(stables)
        min_moves += abs(min_reindeer - min_stables)

        reindeer.pop(reindeer.index(min_reindeer))
        stables.pop(stables.index(min_stables))

    return min_moves
